### PR TITLE
6329 - Remove Internet Explorer

### LINKED ISSUE #6329: Remove support for Internet Explorer (IE)
### Summary

**What we're after:**
_As a web development team, we want to remove support for Internet Explorer (IE) so that we can optimize our codebase, ehance user experience, and leverage modern web technologies without the constraints imposed by an outdated browser._

### Completion criteria

- [x] Update notification banner on the website for IE users informing them of the end of support
- [ ] Update the codebase to remove
   - [x] All IE-specific polyfills and workarounds. 
   - [x] Remove the imports and packages specific for IE
   - [x] Update the CSS and JS to use modern features that are not compatible with IE. (CSS prefixes, babel settings)
   - [x] Remove the X-UA-Compatible tag

### PR DESCRIPTION
**Pull request by rfultz**
## Summary

- Resolves #6329 

We're removing the special rules we've had in place for legacy browsers, especially Internet Explorer.

### Required reviewers

- 1-2 Front-end
- 1 Content (messaging + Draftail functionality)
- 1 UX

## Impacted areas of the application

The entire site, but only for browsers that don't support "modern" features (most of them have been standard for more than a decade).
- updated the language in the banner to make it a little more generic / universal
- renamed the banner's id and updated the style rules for it
- changed the script that loaded the polyfills file so it's switching on the browser banner instead
- removed the polyfills file
- removed the npm entries that were only being used for the polyfills file
- removed the X-UA meta tag

## Screenshots

image

image

## Related PRs

None

## How to test

Content, we can walk you through localhost or we can push to a testing space for you to test the content editor.

- Pull the branch
- `npm i`
- `npm run build`
- `./manage.py runserver`
- Default
   - Load the page
   - Should be no change from Prod (Specifically the very top)
- Banner language and appearance
   - Inspector > Elements
   - Find `<div id="browser_warning" style="display: none;">` near the top, just inside `<body>`
   - Double-click `display: none;` and change `none` to `block` (return)
   - The black banner should appear at the top of the screen, above the blue "LOCAL" etc banner
   - The banner should look like the attached screenshots
- Code (mostly for devs)
   - Inspector > Elements
   - cmd-f for `browserIsCapable`
   - Quickly review that the `<script>` tag doesn't include template comments
- Functionality (mostly for devs)
   - Choose a page template (preferably one for .html and one for .jinja)
   - Find and edit its `var browserIsCapable =` line to something that will fail, like `'fetchgibberish' in window`
   - Reload the page (may need to `npm run build-js`)
   - The black banner should appear at the top of the screen _and_ the window should scroll to show it (may not scroll on a reload, but shouldn't let an initial load open the banner upward)
- Content editor
   - The localhost, etc., content editor should load and functionality correctly (draftail was using the polyfills file, too)

Pages to test, various widths:

1. Homepage
1. About & other Wagtail pages (H4CC, etc)
1. Calendar
1. Data pages ( /data/* )
1. Raising by the numbers (aggregate totals, contributions map)
1. Legal pages

### CODE REVIEW

**Review by johnnyporkchops — CHANGES_REQUESTED** [review]
Followed the helpful how-to-test (thanks for that!) and I everything worked except for templates  extending main.jinja

File: fec/data/templates/layouts/main.jinja [hunk 1]
```diff
@@ -15,13 +15,11 @@
...
-    var canSkipPolyfills = ('fetch' in window && 'assign' in Object);
-    if (!canSkipPolyfills) {
-      var pfScriptElem = document.createElement('script');
-      pfScriptElem.async = false;
-      pfScriptElem.src = '{{ asset_for_js('polyfills.js') }}';
-      document.head.appendChild(pfScriptElem);
+  <script>{# Currently checking for window.fetch and Object.assign #}
+    var browserIsCapable = ('fetch' in window && 'assign' in Object);
+    if (!browserIsCapable) {
+      var browserWarningElement = document.getElementById('browser_warning');
```
[the remark this answers is not in the document]
Comment (rfultz) [reply fec/data/templates/layouts/main.jinja]: Nice catch! I've moved all of those `<script>` to right after the `#browser_warning` elements

**Comment by JonellaCulmer** [comment]
@rfultz I think we can be a bit more helpful and redirect folks to a browser that's more like to work. If you're on an older browser like that, you might not be hip to modern browsers. Can we adjust the language a little bit more? And if we're saying that this message will only appear on IE and if IE is not supported at all, we shouldn't say upgrade IE. Also tagging some content folks for their thoughts on the language;.

Suggested text change: 
Your web browser is not supported
This site uses features that are not supported by this browser. Please upgrade or switch to another browser for a better experience, **such as a recent version of Chrome, Edge, or Firefox.**

cc: @kathycarothers @djgarr @patphongs

**Comment by rfultz** [comment]
@JonellaCulmer updates: (the underline is the hover for that link)
image

**Comment by djgarr** [comment]
I think we need to remain neutral on other browsers people could use for the site and not identify specific ones to switch to. I'd recommend removing everything after "switch to another browser"

**Comment by rfultz** [comment]
@JonellaCulmer @djgarr What about linking to something like https://browser-update.org ?

**Comment by djgarr** [comment]
@rfultz If we link to anything externally, I think it should only be to another government website

**Comment by rfultz** [comment]
@djgarr I can't seem to find where other agencies are linking elsewhere, but [USA.gov](https://www.usa.gov/will-your-computer-work), [USAJobs](https://help.usajobs.gov/faq/troubleshoot), and the [Census Bureau](https://www.census.gov/about/www/browser-requirements.html) all list required browsers by name and some link to them

**Review by johnnyporkchops — APPROVED** [review]
Looks good, thanks @rfultz

**Review by patphongs — APPROVED** [review]

**Comment by JonellaCulmer** [comment]
@rfultz Is this PR still needed?

**Pull request merged**